cpp/tests/io/json_test.cpp

```diff
@@ -2084,71 +2087,144 @@ TEST_F(JsonReaderTest, MixedTypes)
         .lines(true);
 
     cudf::io::table_with_metadata result = cudf::io::read_json(in_options);
+    static int num_case                  = 0;
+    num_case++;
+    std::cout << "case:" << num_case << "\n";
     CUDF_TEST_EXPECT_COLUMNS_EQUAL(result.tbl->get_column(0), expected);
   };
-
-  // test cases.
+  // value + string (not mixed type case)
   test_fn(R"(
 { "a": "123" }
 { "a": 123 }
 )",
           cudf::test::strings_column_wrapper({"123", "123"}));
 
+  // test cases.
+  // STR + STRUCT, STR + LIST, STR + null
+  // STRUCT + STR, STRUCT + LIST, STRUCT + null
+  // LIST + STR, LIST + STRUCT, LIST + null
+  // LIST + STRUCT + STR, STRUCT + LIST + STR, STR + STRUCT + LIST, STRUCT + LIST + null
+  // STR + STRUCT + LIST + null
+
+  // STRING mixed:
+  // STR + STRUCT, STR + LIST, STR + null
   test_fn(R"(
-{ "a": [1,2,3] }
+{ "a": "123" }
 { "a": { "b": 1 } }
 )",
-          cudf::test::strings_column_wrapper({"[1,2,3]", "{ \"b\": 1 }"}));
+          cudf::test::strings_column_wrapper({"123", "{ \"b\": 1 }"}));
+  test_fn(R"(
+{ "a": "123" }
+{ "a": [1,2,3] }
+)",
+          cudf::test::strings_column_wrapper({"123", "[1,2,3]"}));
+  test_fn(R"(
+{ "a": "123" }
+{ "a": null }
+)",
+          cudf::test::strings_column_wrapper({"123", ""}, std::vector<bool>{1, 0}.begin()));
 
+  // STRUCT mixed:
+  // STRUCT + STR, STRUCT + LIST, STRUCT + null
   test_fn(R"(
+{ "a": { "b": 1 } }
 { "a": "fox" }
+)",
+          cudf::test::strings_column_wrapper({"{ \"b\": 1 }", "fox"}));
+  test_fn(R"(
+{ "a": { "b": 1 } }
+{ "a": [1,2,3] }
+)",
+          cudf::test::strings_column_wrapper({"{ \"b\": 1 }", "[1,2,3]"}));
+  cudf::test::fixed_width_column_wrapper<int64_t> child_int_col_wrapper{1, 2};
+  test_fn(R"(
 { "a": { "b": 1 } }
+{ "a": null }
 )",
-          cudf::test::strings_column_wrapper({"fox", "{ \"b\": 1 }"}));
+          cudf::test::structs_column_wrapper{
+            {child_int_col_wrapper}, {1, 0} /*Validity*/
+          });
 
+  // LIST mixed:
+  // LIST + STR, LIST + STRUCT, LIST + null
   test_fn(R"(
 { "a": [1,2,3] }
-{ "a": "fox" }
+{ "a": 123 }
+)",
+          cudf::test::strings_column_wrapper({"[1,2,3]", "123"}));
+  test_fn(R"(
+{ "a": [1,2,3] }
+{ "a": { "b": 1 } }
+)",
+          cudf::test::strings_column_wrapper({"[1,2,3]", "{ \"b\": 1 }"}));
+  test_fn(
+    R"(
+{ "a": [1,2,3] }
+{ "a": null }
+)",
+    cudf::test::lists_column_wrapper{{LCWI{1L, 2L, 3L}, LCWI{4L, 5L}}, valid_t{1, 0}.begin()});
+
+  // All mixed:
```

Comment:
Similar to the comment above - should we consider all permutations of `LIST, STRUCT, STR, NULL`?

Reply:
not required. It's only the mixed types order that needs to be changed.